services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/docker_api/_core.py

```diff
@@ -135,6 +135,29 @@ async def get_dynamic_sidecars_to_observe(
     ]
 
 
+async def _get_service_latest_task(service_id: str) -> Mapping[str, Any]:
+    try:
+        async with docker_client() as client:
+            running_services = await client.tasks.list(
+                filters={"service": f"{service_id}"}
+            )
+            if not running_services:
+                raise DockerServiceNotFoundError(service_id=service_id)
+
+            # The service might have more then one task because the
+            # previous might have died out.
+            # Only interested in the latest task as only one task per
+            # service will be running.
+            sorted_tasks = sorted(running_services, key=lambda task: task["UpdatedAt"])
```

Comment:
``task["UpdatedAt"]`` returns a string with the timestamp ... don't you need to create a ``datetime`` to have a proper sort?

More like ... (to be checked)
```python
sorted(running_services, key=lambda task: datetime(task["UpdatedAt"]) )
```

Reply:
hmm, that could be, I re-used code that was already in that module. will check